core/modules/file/file.module

```diff
@@ -1306,6 +1307,28 @@ function file_managed_file_process($element, &$form_state, $form) {
     '#attributes' => array('class' => array('file-remove-button')),
   );
 
+  // Add a button that will be used to select existing files if enabled.
+  $element['browse_button'] = array(
+    '#name' => implode('_', $element['#parents']) . '_browse_button',
+    '#type' => 'submit',
```

Comment:
Can we make the type `button` here instead? It's not really submitting a form, is it?

Reply:
I think it has to be `submit` in order to submit the form data? But the other buttons in file.module are also "submit", such as the "Remove" button and "Upload" button. I think for styling consistency, using the same type everywhere will prevent discrepancies. Though this button doesn't do anything without JS... I'll investigate if changing it will work.

Reply:
So it looks like whether `#type` is `button` or `submit`, the markup is the same. `theme_button()` specifically sets `$element['#attributes']['type'] = 'submit';`

It works both ways, and the markup is identical. I think I'd prefer to leave it's type as "submit" since that's more commonly used and matches the other buttons in this element.

Reply:
Hah :) okay, disregard.